src/python/gennylib/auto_tasks.py

```diff
@@ -0,0 +1,361 @@
+            # TODO: don't use rtimmons/
+            "$(git merge-base HEAD rtimmons/master) -- src/workloads/"
+        )
+        print(f"Command: {command}")
+        lines = _check_output(self.repo_root, command, shell=True)
+        return {os.path.join(self.repo_root, line) for line in lines if line.endswith(".yml")}
+
+
+class OpName(enum.Enum):
+    ALL_TASKS = object()
+    VARIANT_TASKS = object()
+    PATCH_TASKS = object()
+
+
+class CLIOperation(NamedTuple):
+    mode: OpName
+    variant: Optional[str]
+    is_legacy: bool
+    output_file_suffix: str
+
+    @property
+    def repo_root(self) -> str:
+        if self.is_legacy:
+            if self.mode != OpName.VARIANT_TASKS:
+                return "./genny/genny"
+            return "../src/genny/genny"
+        return "./src/genny"
+
+    @property
+    def output_file(self) -> str:
+        if self.is_legacy:
+            return os.path.join(self.repo_root, self.output_file_suffix)
+        return self.output_file_suffix
+
+    @staticmethod
+    def parse(argv: List[str]) -> "CLIOperation":
+        mode = OpName.ALL_TASKS
+        is_legacy = False
+        variant = None
+        output_file = None
+
+        if "--generate-all-tasks" in argv:
+            mode = OpName.ALL_TASKS
+            is_legacy = True
+        if "--modified" in argv:
+            mode = OpName.PATCH_TASKS
+            is_legacy = True
+        if "--autorun" in argv:
+            mode = OpName.VARIANT_TASKS
+            is_legacy = True
+        if mode in {OpName.PATCH_TASKS, OpName.VARIANT_TASKS}:
+            variant = argv.index("--variants")
+            variant = argv[variant + 1]
+        if "--output" in argv:
+            output_file = argv[argv.index("--output") + 1]
+
+        if is_legacy is False:
+            output_file = "./run/build/Tasks/Tasks.json"
+            if argv[1] == "all_tasks":
+                mode = OpName.ALL_TASKS
+            if argv[1] == "patch_tasks":
+                mode = OpName.PATCH_TASKS
+            if argv[1] == "variant_tasks":
+                mode = OpName.VARIANT_TASKS
+            if not os.path.exists("expansions.yml"):
+                raise Exception(f"No expansions.yml in cwd={os.getcwd()}")
+            with open("expansions.yml") as exp:
+                parsed = yaml.safe_load(exp)
+                variant = parsed["build_variant"]
+        return CLIOperation(mode, variant, is_legacy, output_file)
+
+
+class Runtime:
+    use_expansions_yml: bool = False
+
+    def __init__(self, cwd: str, conts: Optional[dict] = None):
+        self.conts = conts
+        self.cwd = cwd
+
+    def _load(self):
+        if self.conts is not None:
+            return
+        conts = _yaml_load(
+            [os.path.join(self.cwd, f"{b}.yml") for b in {"bootstrap", "runtime", "expansions"}]
+        )
+        if "expansions" in conts:
+            self.use_expansions_yml = True
+            conts = conts["expansions"]
+        else:
```

Comment:
Will the else-clause go away eventually?

Reply:
Yessir! We'll just assert `expansions.yml` exists.